### PR TITLE
feat: client docs

### LINKED ISSUE #7: Document API client
Flesh out the [README](https://github.com/ipfs-shipyard/nft.storage/blob/main/client/Readme.md) so that there's enough information for users to be able to install and use the client and all it's methods. Add CI/dep/codecov badges, license info in standard readme style.

### LINKED ISSUE #8: Add linter to API client
For some code consistency and catching some programming errors. Run in CI.

### LINKED ISSUE #9: Add repository, bugs and homepage to package.json
So that users can access the code and website from the listing on the npm website.

### PR DESCRIPTION
**Pull request by Gozala**
- [x] fixes #9
- [x] fixes #8
- [x] fixes #7

### CODE REVIEW

**Comment by Gozala** [comment]
hmm, I may have broke github action

File: client/src/lib.js [hunk 1]
```diff
@@ -1,22 +1,64 @@
-import * as API from "./api.js"
+/**
+ * A client library for https://nft.storage/ service. It provides convenient
+ * interface for working with [RAW HTTP API](https://nft.storage/#api-docs)
+ * from web browser or [nodejs](https://nodejs.org/) and comes bundled with
+ * TS for out-of-the box type inference and better IntelliSense.
```
Comment (alanshaw) [inline client/src/lib.js:5]:
```suggestion
 * A client library for the https://nft.storage/ service. It provides a convenient
 * interface for working with the [Raw HTTP API](https://nft.storage/#api-docs)
 * from a web browser or [Node.js](https://nodejs.org/) and comes bundled with
 * TS for out-of-the box type inference and better IntelliSense.
```

File: client/src/lib.js [hunk 2]
```diff
@@ -112,34 +156,78 @@ export default class NFTStore {
   // Just a sugar so you don't have to pass around endpoint and token around.
 
   /**
+   * Stores a single file and returns a corresponding Content Identifier (CID).
+   * Takes a [Blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob)
+   * or a [File](https://developer.mozilla.org/en-US/docs/Web/API/File). Note
+   * that no file name or it's metadata is retained.
```
Comment (alanshaw) [inline client/src/lib.js:162]:
```suggestion
   * Stores a single file and returns the corresponding Content Identifier (CID).
   * Takes a [Blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob)
   * or a [File](https://developer.mozilla.org/en-US/docs/Web/API/File). Note
   * that no file name or file metadata is retained.
```

File: client/src/lib.js [hunk 3]
```diff
@@ -112,34 +156,78 @@ export default class NFTStore {
...
+   * ```
+   *
    * @param {Blob} blob 
    */
   storeBlob(blob) {
-    return NFTStore.storeBlob(this, blob)
+    return NFTStorage.storeBlob(this, blob)
   }
   /**
+   * Stores a directory of files and returns a CID for a directory.
```
Comment (alanshaw) [inline client/src/lib.js:177]:
```suggestion
   * Stores a directory of files and returns a CID for the directory.
```

File: client/src/lib.js [hunk 4]
```diff
@@ -112,34 +156,78 @@ export default class NFTStore {
...
    */
   storeDirectory(files) {
-    return NFTStore.storeDirectory(this, files)
+    return NFTStorage.storeDirectory(this, files)
   }
   /**
+   * Returns current status of the stored content by its CID.
+   * @example
+   * ```js
+   * const statu = await client.status('Qmf412jQZiuVUtdgnB36FXFX7xg5V6KEbSJ4dpQuhkLyfD')
```
Comment (alanshaw) [inline client/src/lib.js:199]:
```suggestion
   * const status = await client.status('Qmf412jQZiuVUtdgnB36FXFX7xg5V6KEbSJ4dpQuhkLyfD')
```

File: client/README.md [hunk 5]
```diff
@@ -1,16 +1,33 @@
...
 
-### Local development
+A client library for https://nft.storage/ service. It provides convenient interface for working with [RAW HTTP API][] from web browser or [nodejs][] and comes bundled with TS for out-of-the box type inference and better IntelliSense.
 
-```bash
-yarn install
-yarn start
+## Install
+
+Install the package from npm using using npm
```
Comment (alanshaw) [inline client/README.md:7]:
```suggestion
Install the package using npm
```

File: client/README.md [hunk 6]
```diff
@@ -1,16 +1,33 @@
-# API Client
+# nft.storage
 
-### Local development
+A client library for https://nft.storage/ service. It provides convenient interface for working with [RAW HTTP API][] from web browser or [nodejs][] and comes bundled with TS for out-of-the box type inference and better IntelliSense.
```
Comment (alanshaw) [inline client/README.md:3]:
```suggestion
A client library for the https://nft.storage/ service. It provides a convenient interface for working with the [Raw HTTP API][] from a web browser or [Node.js][] and comes bundled with TS for out-of-the box type inference and better IntelliSense.
```

File: client/README.md [hunk 7]
```diff
@@ -1,16 +1,33 @@
...
+Or yarn
 
 ```
-yarn run build
-yarn publish
+yarn add nft.storage
 ```
 
+
+### Usage
```
Comment (alanshaw) [inline client/README.md:20]:
```suggestion
### Usage

First, obtain an API token from https://nft.storage and use it in place of `API_TOKEN` below:
```

File: client/README.md [hunk 8]
```diff
@@ -1,16 +1,33 @@
...
+```js
+import { NFTStorage } from "nft.storage"
+const client = new NFTStorage({ token: API_TOKEN }) 
+const cid = await client.storeBlob(new Blob(['hello world'])) 
+```
+
+For more examples please see [API documentation][]
+
+
+[RAW HTTP API]:https://nft.storage/#api-docs
```
Comment (alanshaw) [inline client/README.md:31]:
```suggestion
[Raw HTTP API]:https://nft.storage/#api-docs
```

File: client/README.md [hunk 9]
```diff
@@ -1,16 +1,33 @@
...
+[nodejs]:https://nodejs.org/
```
Comment (alanshaw) [inline client/README.md:32]:
```suggestion
[Node.js]:https://nodejs.org/
```

**Review by alanshaw — CHANGES_REQUESTED** [review]
It might be better to use `bafy...` CID examples as that is what the user will see.

**Pull request merged**